pallets/confidential-asset/src/lib.rs

```diff
@@ -27,30 +27,25 @@ use cryptography::{
     AssetId,
 };
 use frame_support::{decl_error, decl_event, decl_module, decl_storage, dispatch::DispatchResult};
-use frame_system::{self as system, ensure_signed};
+use frame_system::ensure_signed;
 use pallet_identity as identity;
-use pallet_statistics::{self as statistics};
-use polymesh_common_utilities::Context;
 use polymesh_common_utilities::{
-    balances::Trait as BalancesTrait, identity::Trait as IdentityTrait,
+    asset::Trait as AssetTrait, balances::Trait as BalancesTrait, identity::Trait as IdentityTrait,
+    CommonTrait, Context,
+};
+use polymesh_primitives::{
+    AssetIdentifier, AssetName, AssetType, FundingRoundName, IdentifierType, IdentityId, Ticker,
 };
-use polymesh_primitives::IdentityId;
 use polymesh_primitives_derive::VecU8StrongTyped;
-use sp_std::prelude::*;
-
-type Identity<T> = identity::Module<T>;
+use sp_std::{
+    convert::{From, TryFrom},
+    prelude::*,
+};
 
 /// The module's configuration trait.
-pub trait Trait:
-    frame_system::Trait
-    + BalancesTrait
-    + IdentityTrait
-    + pallet_session::Trait
-    + statistics::Trait
-    + pallet_contracts::Trait
-    + pallet_portfolio::Trait
-{
-    type Event: From<Event> + Into<<Self as frame_system::Trait>::Event>;
+pub trait Trait: frame_system::Trait + IdentityTrait + BalancesTrait {
+    type Asset: AssetTrait<Self::Balance, Self::AccountId>;
```

Comment:
In general it would be good to have doc comments both on type parameters and the trait itself.